src/pde/pde_collisional_landau.hpp

```diff
@@ -0,0 +1,495 @@
+    std::transform(x.begin(), x.end(), f.begin(),
+                   [](P const &x_v) -> P { return std::pow(x_v, 2); });
+    return f;
+  }
+
+  inline static moment<P> const moment0 = moment<P>(
+      std::vector<md_func_type<P>>({{moment0_f1, moment0_f1, moment0_f1}}));
+  inline static moment<P> const moment1 = moment<P>(
+      std::vector<md_func_type<P>>({{moment0_f1, moment1_f1, moment0_f1}}));
+  inline static moment<P> const moment2 = moment<P>(
+      std::vector<md_func_type<P>>({{moment0_f1, moment2_f1, moment0_f1}}));
+
+  inline static std::vector<moment<P>> const moments_ = {moment0, moment1,
+                                                         moment2};
+
+  /* Construct (n, u, theta) */
+  // n = density
+  // u = bulk velocity
+  // theta = temperature
+  static P n(P const &x, P const t = 0)
+  {
+    ignore(t);
+
+    return (1.0 + A * std::cos(0.5 * x));
+  }
+
+  static P u(P const &x, P const t = 0)
+  {
+    ignore(t);
+    ignore(x);
+    return 0.0;
+  }
+
+  static P theta(P const &x, P const t = 0)
+  {
+    ignore(t);
+    ignore(x);
+    return 1.0;
+  }
+
+  // E = -d_x phi
+  static P E(P const &x, P const t = 0)
+  {
+    ignore(t);
+    ignore(x);
+    return 0.0;
+  }
+
+  // source for poisson problem is -d_xx phi = n - 1 = S(n)
+  static P S(P const &y, P const t = 0)
+  {
+    ignore(t);
+    // subtracts quadrature values by one
+    return y - 1.0;
+  }
+
+  // holds the maximum absolute value of E
+  static P MaxAbsE(P const &x, P const t = 0)
+  {
+    ignore(t);
+    ignore(x);
+    return 0.0;
+  }
+
+  /* build the terms */
+
+  // Term 1
+  // -v\cdot\grad_x f for v > 0
+  //
+  static P e1_g1(P const x, P const time = 0)
+  {
+    ignore(x);
+    ignore(time);
+    return -1.0;
+  }
+
+  static P e1_g2(P const x, P const time = 0)
+  {
+    ignore(time);
+    return (x > 0.0) ? x : 0.0;
+  }
+
+  inline static const partial_term<P> e1_pterm_x = partial_term<P>(
+      coefficient_type::div, e1_g1, nullptr, flux_type::downwind,
+      boundary_condition::periodic, boundary_condition::periodic);
+
+  inline static const partial_term<P> e1_pterm_v = partial_term<P>(
+      coefficient_type::mass, e1_g2, nullptr, flux_type::central,
+      boundary_condition::periodic, boundary_condition::periodic);
+
+  inline static term<P> const term_e1x =
+      term<P>(false,  // time-dependent
+              "E1_x", // name
+              {e1_pterm_x}, imex_flag::imex_explicit);
+
+  inline static term<P> const term_e1v =
+      term<P>(false,  // time-dependent
+              "E1_v", // name
+              {e1_pterm_v}, imex_flag::imex_explicit);
+
+  inline static std::vector<term<P>> const terms_1 = {term_e1x, term_e1v};
+
+  // Term 2
+  // -v\cdot\grad_x f for v < 0
+  //
+  static P e2_g1(P const x, P const time = 0)
+  {
+    ignore(x);
+    ignore(time);
+    return -1.0;
+  }
+
+  static P e2_g2(P const x, P const time = 0)
+  {
+    ignore(time);
+    return (x < 0.0) ? x : 0.0;
```

Comment:
Is it better to say `return std::min(x, 0.0);`

Not sure, your call.

Reply:
std::min requires `x` and `0.0` to be the same type. Something like the suggestion below should work.
```suggestion
    return std::min(x, P{0.0});
```

Reply:
This is already done in the next PR: https://github.com/project-asgard/asgard/pull/512/commits/1d913b57a8c08cf8634f6d90586932eb898be361

Reply:
We should not have let these linger for so long.